src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/StringSearchValues.cs

```diff
@@ -248,6 +251,18 @@ static SearchValues<string> PickAhoCorasickImplementation<TCaseSensitivity>(AhoC
 
             Debug.Assert(!(asciiStartLettersOnly && asciiStartUnaffectedByCaseConversion));
 
+            // If we still have empty buckets we could use and we're ignoring case, we may be able to
+            // generate all possible permutations of the first N characters and switch to case-sensitive searching.
+            // E.g. ["ab", "c!"] => ["ab", "Ab" "aB", "AB", "c!", "C!"].
```

Comment:
It would also handle "ab" /"cd", right? I'm curious about the choice to use! I'm the example.

Reply:
It would handle that case, but that'd already put it at the limit of 8 values. Anything more and it'd be too much.

I used a symbol since that's more likely to actually hit this case in practice given the fewer resulting permutations.